File: valid_anagram/valid_anagram.rb
s = "rat"
t= "tar"

def is_anagram(s, t)
  if s.size != t.size
      return false
  end
  #hash tables
  s_table = Hash.new(0)
  t_table = Hash.new(0)

  #arrays
  s_arr = s.split("")
  t_arr = t.split("")

  s_arr.each do | char |
      s_table[char] += 1
  end 
end

File: valid_anagram/valid_anagram.js
const s = "rat"
const t = "tart"

var isAnagram = (s, t) => {
  if (s.length !== t.length) {
      return false
  }
  const s_obj = {}
  const t_obj = {}

  for (let i = 0; i < s.length; i++) {
      if (!s_obj[s[i]]) {
        s_obj[s[i]] = 1;  
      } else {
        s_obj[s[i]]++;
      };
  }
//   for (i = 0; i < t.length; i++) {
//     if (!t_obj[t[i]]) {
//       t_obj[t[i]] = 1  
//     } else {
//       t_obj[t[i]]++
//     }
// }
for (let i = 0; i < t.length; i ++) {
  if (s_obj[t[i]]){
    s_obj[t[i]]--;
  } else {
    return false
  }
}

if (Object.values(s_obj).every((v) => v === 0)){
  return true
}
  
};

console.log(isAnagram(s, t))